src/inmanta/module.py

```diff
@@ -3022,13 +3026,15 @@ def get_plugin_files(self) -> Iterator[tuple[Path, ModuleName]]:
         ):
             raise InvalidModuleException(f"Directory {plugin_dir} should be a valid python package with a __init__.py file")
 
-        return (
+        self._plugin_file_cache = [
             (
                 Path(file_name),
                 ModuleName(self._get_fq_mod_name_for_py_file(file_name, plugin_dir, self.name)),
             )
             for file_name in self._list_python_files(plugin_dir)
-        )
+        ]
+
+        return iter(self._plugin_file_cache)
```

Comment:
I would love to change the signature of this method, but it is stable api, ...

Reply:
because we don't really need it to return the module name and it to be an iterator?

Reply:
we don't need it to be an iterator mostly